src/io/flutter/run/daemon/FlutterApp.java

```diff
@@ -68,7 +68,8 @@
   private final @NotNull Project myProject;
   private final @Nullable Module myModule;
   private final @NotNull RunMode myMode;
-  private final @NotNull FlutterDevice myDevice;
+  // TODO(github.com/flutter/flutter-intellij/issues/3293) myDevice is not-null for all run configurations except flutter web configurations
```

Comment:
The bug listed doesn't cover the issue. e.g. whether Device should be null for FlutterWeb.